perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/MembersManagerMethod.java

```diff
@@ -126,7 +125,12 @@ public RichMember call(ApiCaller ac, Deserializer params) throws PerunException
 			String password = params.readString("password");
 			Vo vo =  ac.getVoById(params.readInt("vo"));
 			String namespace = params.readString("namespace");
-			User sponsor = ac.getUsersManager().getUserByExtSourceNameAndExtLogin(ac.getSession(), params.readString("extSourceName"), params.readString("extLogin"));
+			User sponsor;
+			try {
+				sponsor = ac.getUserById(params.readInt("sponsor"));
+			} catch (RpcException exp) {
```

Comment:
You can use params.contains("sponsor") to check if the parameter is available. Using that, you don't have to deal with the exception anymore.